### PR TITLE
Bl-6441 UI - additional http code

### PR DESCRIPTION
**Pull request by Marcin-Duszynski**
### What are the relevant Jira tickets?
Bl-6441

### Where should the reviewer start?
main.js
README.md

New http code added: 422

## Dev checklist:
- Is proper destination **branch** selected?
- Are proper **label** and **reviewers** selected?
- Has code been tested manually?
- Has code been covered by unit tests?
- Are unit test executed on Jenkins during pull request build?
- Do all unit tests pass?
- Is Jira ticket state updated?

## Code review checklist (for reviewers -> Copy to your comment)
- Is proper destination **branch** selected?
- Are proper **label** and **reviewers** selected?
- Do you understand task goal and business context?
- Is documentation updated and understandable?
- Has code been covered by unit tests?
- Did you identify technical issues/debt and reported it to TA? 
- Does code meet SOLID principles?
- Does code meet DRY principle?
- Is code in better state than before (scout rule)?

### CODE REVIEW

File: app/src/main.js [hunk 1]
```diff
@@ -7,33 +7,57 @@ const serviceConfig = require('./config/service');
...
+  if (recall.success) {
+    logger.info({ context: { make, vin, recall } }, 'Recall fetched successfully.');
+    res.status(200)
+      .send({
+        status_description: recall.description,
+        vin_recall_status: recall.status,
+        last_update: recall.lastUpdate,
+      });
+  } else if (recall.actionCode === responseCode.smmtInvalidMarque) {
+    const context = {
```
Comment (ryszardm-kainos) [inline app/src/main.js:26]: there is quite a bit of code replication in those else-ifs. Is it feasible to extract the content of those into a parametrised function?

File: app/src/main.js [hunk 2]
```diff
@@ -7,33 +7,57 @@ const serviceConfig = require('./config/service');
...
+      make, vin, errors: recall.errors,
+    };
+
+    logger.error({ context }, 'Recall fetching from SMMT failed. Invalid make.');
+    res.status(422).send({
```
Comment (ryszardm-kainos) [inline app/src/main.js:31]: Think it would be nice to document the new http status behaviour in our Readme

Comment (Marcin-Duszynski) [reply app/src/main.js:31]: I forgot to push updated doc. done

File: app/src/main.js (hunk 1 above)
Comment (Marcin-Duszynski) [reply app/src/main.js:26]: discussed f2f

**Review by ryszardm-kainos — APPROVED** [review]

**Pull request merged**